include/ews/ews.hpp

```diff
@@ -11494,15 +12727,10 @@ namespace ews
             throw exception("Unknown property path");
         }
 
-    private:
+    protected:
         std::string uri_;
```

Comment:
Never allow derived classes access to your data-members. Never. Use NVI if you need different behavior in sub-classes. This is what polymorphic classes are for. See https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Non-Virtual_Interface.